osf_tests/test_notable_domains.py

```diff
@@ -0,0 +1,60 @@
+import pytest
+from urllib.parse import urlparse
+
+from osf_tests.factories import (
+    CommentFactory,
+    NodeFactory,
+    PreprintFactory,
+    RegistrationFactory,
+)
+from osf.models import (
+    NotableDomain,
+    SpamStatus
+)
+from osf.external.spam.tasks import check_resource_for_domains
+from framework.celery_tasks.handlers import enqueue_task
+
+
+@pytest.mark.django_db
+class TestNotableDomain:
```

Comment:
Please do a more realistic test of this via the `check_spam` call with domains in SPAM_CHECK_FIELDS (both node and preprint). Also, please confirm that the regex/parsing work as expected by having multiple domains (and multiple entries for the same domain). Also, interactions with `HAM`ed resources/users.